cmd/buildkitapi/main.go

```diff
@@ -41,20 +41,25 @@ func run() error {
 
 	d.NegotiateAPIVersion(ctx)
 
-	archive, err := build.TarPath(ctx, ".")
-	if err != nil {
-		return err
-	}
+	pr, pw := io.Pipe()
+	go func() {
+		err := build.TarPath(ctx, pw, ".")
+		if err != nil {
```

Comment:
you can get rid of this `if` branch. it's worth pointing out that Close() is literally the same as CloseWithError(nil)
https://golang.org/src/io/pipe.go?s=3046:3098#L128